### PR TITLE
Test/refactor

### PR DESCRIPTION
**Pull request by JhChoy**
1. Re-struct the test directory as below:
```
test
 ├── foundry
 │   ├── unit
 │   │   ├── ContractA # Divide unit tests if the contract is large or complicated
 │   │   │   ├── FunctionA.t.sol # Test Contract Name: ContractAFunctionAUnitTest
 │   │   │   │                   # 'UnitTest' suffix must be included
 │   │   │   └── FunctionB.t.sol 
 │   │   └── SimpleContract.t.sol # Just one unit test file for the simple contract
 │   │                            # Test Contract Name: SimpleContractUnitTest
 │   └── integration
 │       ├── ScenarioA.t.sol # Divide into scenarios
 │       │                   # Test Contract Name: 'IntegrationTest' suffix must be included
 │       │                   # e.g.) ScenarioAIntegrationTest
 │       ...
 ├── hardhat # Hardhat tests are not used except in exceptional cases.
 └── mocks # Define mock contracts
     └── MockToken.sol
```

2. Make a SetUp contract for duplicated setup logic of unit tests.
3. Move `Constants.sol` to `/unit`.

### CODE REVIEW

**Comment by detectivekim** [comment]
If you end up changing `SetUp.sol` later, you will also need to change other files. Therefore, I think it would be better to write tests that can be independent, rather than tests that are dependent on the `SetUp.sol` file.

File: contracts/CallOptionToken.sol [hunk 1]
```diff
@@ -47,8 +45,8 @@ contract CallOptionToken is ERC20, CloberOptionToken, ReentrancyGuard, Ownable {
     ) ERC20(name_, symbol_) {
         _underlyingToken = IERC20(underlyingToken_);
         _quoteToken = IERC20(quoteToken_);
-        _quotePrecisionComplement = 10**(18 - IERC20Metadata(quoteToken_).decimals());
-        _underlyingPrecisionComplement = 10**(18 - IERC20Metadata(underlyingToken_).decimals());
+        _precisionComplement =
+            10**(18 - IERC20Metadata(quoteToken_).decimals() + IERC20Metadata(underlyingToken_).decimals());
```
Comment (JhChoy) [inline contracts/CallOptionToken.sol]:
```suggestion
            10**(18 + IERC20Metadata(underlyingToken_).decimals() - IERC20Metadata(quoteToken_).decimals());
```

Comment (JhChoy) [reply contracts/CallOptionToken.sol]: Also, `IERC20Metadata(underlyingToken_).decimals()` is called twice. We can reduce it to one call. See `PutOptionToken`.

File: contracts/CallOptionToken.sol [hunk 2]
```diff
@@ -101,12 +99,12 @@ contract CallOptionToken is ERC20, CloberOptionToken, ReentrancyGuard, Ownable {
...
         require(optionAmount > 0, "INVALID_AMOUNT");
 
-        uint256 collateralAmount = Math.ceilDiv(optionAmount, _underlyingPrecisionComplement);
+        uint256 collateralAmount = optionAmount;
 
         _quoteToken.safeTransferFrom(
             msg.sender,
             address(this),
-            Math.ceilDiv(optionAmount * strikePrice, _quotePrecisionComplement * _PRICE_PRECISION)
+            Math.ceilDiv(optionAmount * strikePrice, _precisionComplement)
```
Comment (JhChoy) [inline contracts/CallOptionToken.sol:107]: This code looks weird. I believe it should be
```suggestion
            Math.ceilDiv(optionAmount * _precisionComplement / strikePrice)
```
Note that `strikePrice` is underlying/quote.

File: contracts/CallOptionToken.sol (hunk 1 above)
Comment (detectivekim) [reply contracts/CallOptionToken.sol]: Resolved

**Review by detectivekim — APPROVED** [review]

**Pull request merged**